tests/MongoDB.Bson.Tests/ObjectModel/ObjectIdTests.cs

```diff
@@ -14,20 +14,76 @@
 */
 
 using System;
+using System.Globalization;
 using System.Linq;
-using MongoDB.Bson;
+using FluentAssertions;
+using MongoDB.Bson.TestHelpers;
 using Xunit;
 
 namespace MongoDB.Bson.Tests
 {
     public class ObjectIdTests
     {
+        [Theory]
+        [InlineData(0xFFFFFE, 0xFFFFFF)]
+        [InlineData(0xFFFFFF, 0)]
+        [InlineData(0x1000000, 1)]
+        [InlineData(0x1FFFFFE, 0xFFFFFF)]
+        [InlineData(0x1FFFFFF, 0)]
+        [InlineData(0x2000000, 1)]
+        public void Ensure_that_counter_is_resetted_after_max_value(int incrementValue, int expectedIncrement)
+        {
+            ObjectIdReflector.__staticIncrement(incrementValue);
+            var oid = ObjectId.GenerateNewId();
+            var resultCounter = oid.Increment;
+            resultCounter.Should().Be(expectedIncrement);
+        }
+
+        [Fact]
+        public void Ensure_that_machine_value_is_not_the_same_if_created_from_different_appdomain()
+        {
+#if NET452
+            var parentOid = ObjectId.GenerateNewId();
+
+            ObjectId childOid = default;
+            AppDomain testDomain = null;
+            try
+            {
+                testDomain = AppDomain.CreateDomain("MyDomain", null, new AppDomainSetup { ApplicationBase = AppDomain.CurrentDomain.SetupInformation.ApplicationBase });
+                var marshalType = typeof(AppDomainCourier);
+                var testProxy = (AppDomainCourier)testDomain.CreateInstanceAndUnwrap(marshalType.Assembly.FullName, marshalType.FullName);
+                childOid = testProxy.ObjectId;
+            }
+            finally
+            {
+                if (testDomain != null)
+                {
+                    AppDomain.Unload(testDomain);
+                }
+            }
+
+            childOid.Machine.Should().NotBe(parentOid.Machine);
+#endif
+        }
```

Comment:
I suggest the following slight renamings and refactoring:

    #if NET452
            [Fact]
            public void Ensure_that_machine_value_is_not_the_same_if_created_from_different_appdomain()
            {
                var objectIdCreatedInThisAppDomain = ObjectId.GenerateNewId();
    
                var differentAppDomain = AppDomain.CreateDomain("DifferentObjectIdCreatorAppDomain", null, new AppDomainSetup { ApplicationBase = AppDomain.CurrentDomain.SetupInformation.ApplicationBase });
                try
                {
                    var objectIdCreatorType = typeof(ObjectIdCreator);
                    var objectIdCreator = (ObjectIdCreator)differentAppDomain.CreateInstanceAndUnwrap(objectIdCreatorType.Assembly.FullName, objectIdCreatorType.FullName);
                    var objectIdCreatedInDifferentAppDomain = objectIdCreator.CreateObjectId(); // CreateObjectId method runs in differentAppDomain
                    objectIdCreatedInDifferentAppDomain.Machine.Should().NotBe(objectIdCreatedInThisAppDomain.Machine);
                }
                finally
                {
                    AppDomain.Unload(differentAppDomain);
                }
            }
    #endif